### PR TITLE
Multiple invitations object found fix

### PR DESCRIPTION
**Pull request by say-yawn**
# About this PR
Currently, we use `get()` on the queryset to retrieve existing `Invitations` which can result into the error stated in #488. 

# Acceptance Criteria
- [ ] handle multiple invitations in  `invite_beta_user`
- [x] handle multiple invitations in `invitations_only` signal
- [x] handle multiple invitations in `waitlist` view
- [x] handle multiple invitations in `invite_monitor_waitlist` command
- [x] keep the oldest invitation and delete new ones

###
## Nice to have:
- [ ] Constrain email and fxa_uid to be unique on `Invitations`?

### CODE REVIEW

File: privaterelay/models.py [hunk 1]
```diff
@@ -14,6 +14,24 @@ def __str__(self):
         return 'Invitation for %s' % self.email
 
 
+def get_invitation(email=None, fxa_uid=None, active=False):
+    invitations = Invitations.objects.filter(
+        Q(email=email) | Q(fxa_uid=fxa_uid), active=active
```
Comment (groovecoder) [inline privaterelay/models.py:19]: NameError at /accounts/fxa/login/callback/

name 'Q' is not defined

File: privaterelay/models.py [hunk 2]
```diff
@@ -14,6 +14,24 @@ def __str__(self):
...
+    ).order_by('date_added')
+    if invitation.count() < 1:
```
Comment (groovecoder) [inline privaterelay/models.py:21]:
```suggestion
    if invitations.count() < 1:
```

File: privaterelay/signals.py [hunk 3]
```diff
@@ -55,8 +55,8 @@ def invitations_only(sender, **kwargs):
         return True
 
     except Invitations.DoesNotExist:
-        waitlist_invite = Invitations.objects.filter(
-            Q(email=email) | Q(fxa_uid=fxa_uid), active=False
+        waitlist_invite = get_invitation(
+            email=email, fxa_uid=fxa_uid, active=False
```
Comment (groovecoder) [inline privaterelay/signals.py:59]: Calling this in the `Inviations.DoesNotExist` block means that `get_invitation` also raises the same `DoesNotExist`.

**Comment by groovecoder** [comment]
Should we update this PR? This has only happened once in prod? https://sentry.prod.mozaws.net/operations/fx-private-relay-prod/issues/8876749/?query=is%3Aunresolved

**Comment by groovecoder** [comment]
Closing this as we haven't seen it happen except the single time. We can bring this back if it becomes an issue again.

**Comment by groovecoder** [comment]
We saw this again with the last round of invitations; re-opening PR.

File: privaterelay/signals.py (hunk 3 above)
Comment (groovecoder) [reply privaterelay/signals.py:58]: Nevermind. The above check is for `active=True`, this one is for `active=False` and is OUTSIDE the `try`.

**Comment by groovecoder** [comment]
Rebased and updated this PR. Regression spot-checks:

* [x] Join wait-list works
* [x] Redeeming a monitor wait-list invitation works
* [x] Redeeming a relay wait-list invitation works

I didn't re-created issues with actual multiple invitation objects found though.

**Comment by groovecoder** [comment]
I staged some duplicate invitation objects and spot-checked all the scenarios here except the `invite_beta_user` command, because we don't plan to use that command anymore between now and open beta.

File: privaterelay/models.py [hunk 4]
```diff
@@ -14,6 +15,27 @@ def __str__(self):
...
+        raise Invitations.DoesNotExist(
+            'get_invitation found no invitation for args: %s' % local_args
+        )
+    invitations = list(invitations)
+    oldest_invitation = invitations.pop(0)
+    for invite in invitations:
+        if oldest_invitation.email != invite.email:
+            oldest_invitation.email = invite.email
+        if oldest_invitation.fxa_uid != invite.fxa_uid:
+            oldest_invitation.fxa_uid = invite.fxa_uid
```
Comment (say-yawn) [inline privaterelay/models.py:33]:
```suggestion
        if invite.email and oldest_invitation.email != invite.email:
            oldest_invitation.email = invite.email
        if  invite.fxa_uid and oldest_invitation.fxa_uid != invite.fxa_uid:
            oldest_invitation.fxa_uid = invite.fxa_uid
```
Check to make sure that the `fxa_uid` and `email` have values and that we are updating the fields only when they have values.

**Comment by groovecoder** [comment]
Merged and spot-checked this on the dev server. The only issue seemed to be that `invite_relay_waitlist` isn't using `get_invitation`, so when I forced a duplicate invitation for someone on the Relay wait-list (via `INSERT` directly in the DB), `invite_relay_waitlist` created a dupe invitation.

But, that user was able to claim their invitation with no problems, and I believe the only dupe invitation records we've seen are from BEFORE the Relay wait-list was open, so I don't expect this to happen in prod at all.

So, I'm going to deploy this in time for today's next invitations.

**Pull request merged**